spharpy/ambi.py

```diff
@@ -0,0 +1,274 @@
+                             'for N = {n_max}.')
+
+        self._n_max = n_max
+
+        if basis_type == 'complex' and not is_complex:
+            raise ValueError('Data are real-valued while '
+                             'spherical harmonics bases are complex-valued.')
+
+        self._basis_type = basis_type
+        self._condon_shortley = condon_shortley
+
+        self.normalization = normalization
+        self.channel_convention = channel_convention
+
+        Signal.__init__(self, data, sampling_rate=sampling_rate,
+                        n_samples=n_samples, domain=domain, fft_norm=fft_norm,
+                        comment=comment, is_complex=is_complex)
+
+    @property
+    def n_max(self):
+        return self.n_max
+
+    @property
+    def basis_type(self):
+        return self._basis_type
+
+    @property
+    def normalization(self):
+        return self._normalization
+
+    @normalization.setter
+    def normalization(self, value):
+        if self.normalization is not value:
+            self._recalculate_normalization(self, value)
+
+        self._normalization = value
+
+    @property
+    def channel_convention(self):
+        return self._channel_convention
+
+    @channel_convention.setter
+    def channel_convention(self, value):
+        if self.channel_convention is not value:
+            self._recalculate_channel_convention(value)
+
+    @property
+    def condon_shortley(self):
+        return self._condon_shortley
+
+    def _recalculate_normalization(self, normalization):
+        n_coeff = (self.n_max + 1) ** 2
+
+        for acn in range(n_coeff):
+            if self.channel_convention == "fuma":
+                order, degree = SphericalHarmonics.fuma_to_nm(acn)
+            else:
+                order, degree = SphericalHarmonics.acn_to_nm(acn)
+
+            if self._normalization == 'n3d':
+                if normalization == "sn3d":
+                    self._data[:, acn, ...] *= \
+                        SphericalHarmonics.n3d_to_sn3d_norm(degree, order)
+                elif normalization == "maxN":
+                    self._data[:, acn, ...] *= \
+                        SphericalHarmonics.n3d_to_maxn(acn)
+            if self._normalization == 'sn3d':
+                if normalization == 'n3d':
+                    self._data[:, acn, :] *= \
+                        SphericalHarmonics.sn3d_to_n3d_norm(degree, order)
+                elif normalization == "maxN":
+                    self._data[:, acn, :] *= \
+                        SphericalHarmonics.sn3d_to_maxN(acn)
+            if self._normalization == 'maxN':
+                if normalization == 'n3d':
+                    self._data[:, acn, :] *= \
+                        SphericalHarmonics.maxN_to_n3d(acn)
+                elif normalization == "sn3d":
+                    self._data[:, acn, :] *= \
+                        SphericalHarmonics.maxN_to_sn3d(acn)
+
+    def _recalculate_channel_convention(self, value):
```

Comment:
```suggestion
    def _change_channel_convention(self, value):
```
Just to make it more explicit.